float64 encoder1
float64 encoder2
float64 encoder3
float64 encoder4
float64 encoder5
float64 encoder6